exist-core/src/main/java/org/exist/xquery/functions/map/MapType.java

```diff
@@ -156,7 +156,7 @@ public void add(final AbstractMapType other) {
     public AbstractMapType merge(final Iterable<AbstractMapType> others) {
 
         // create a transient map
-        IMap<AtomicValue, Sequence> newMap = map.linear();
+        IMap<AtomicValue, Sequence> newMap = map.forked();
```

Comment:
ah ok. Being not at home, I could not see the functionality behind `linear()`

Reply:
I don't understand why forked is now used here, as `forked` is called below.

Reply:
That is too late, as the tests clearly prove. But maybe the second call to forked can be removed or changed to `.linear()`?

Reply:
The second call will likely not do much the documentation states:
"For already forked maps the call to .forked() will just return itself".